WireDomain/Sources/WireDomain/Repositories/Message/MessageLocalStore.swift

```diff
@@ -0,0 +1,383 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireDataModel
+
+// sourcery: AutoMockable
+/// Facilitate access to message related domain objects.
+public protocol MessageLocalStoreProtocol {
+
+    func addSystemMessageToConversation(
+        messageType: MessageType,
+        conversationID: UUID,
+        conversationDomain: String?
+    ) async
+
+}
+
+public final class MessageLocalStore: MessageLocalStoreProtocol {
+
+    // MARK: - Properties
+
+    let context: NSManagedObjectContext
+    let conversationLocalStore: any ConversationLocalStoreProtocol
```

Comment:
Here we interact with the conversation local store directly, to fetch a given conversation, I figured it's fine since these local local stores operate on the same level (storage layer), so I'd say we can allow them to do so. What do you think ?